packages/server/tests/acceptance/estimateGasPrecompile.spec.ts

```diff
@@ -0,0 +1,473 @@
+    baseGasCheck(estimateGasResponse, 786166);
+  });
+
+  //EGP-003
+  it('should call estimateGas with dissociate token function without association for fungible token - negative', async function () {
+    const tx = await contract.dissociateTokenExternal.populateTransaction(
+      prefix + EstimatePrecompileContractAddress,
+      tokenAddress,
+    );
+    negativeScenarioVerification(tx, SERVER_ERROR);
+  });
+
+  //EGP-004
+  it('should call estimateGas with dissociate token function without association for NFT - negative', async function () {
+    const tx = await contract.dissociateTokenExternal.populateTransaction(
+      prefix + EstimatePrecompileContractAddress,
+      nftAddress,
+    );
+    negativeScenarioVerification(tx, SERVER_ERROR);
+  });
+
+  //EGP-005
+  it('should call estimateGas with nested associate function that executes it twice for fungible token - negative', async function () {
+    const tx = await contract.nestedAssociateTokenExternal.populateTransaction(
+      prefix + EstimatePrecompileContractAddress,
+      tokenAddress,
+    );
+    negativeScenarioVerification(tx, SERVER_ERROR);
+  });
+
+  //EGP-006
+  it('should call estimateGas with nested associate function that executes it twice for NFT - negative', async function () {
+    const tx = await contract.nestedAssociateTokenExternal.populateTransaction(
+      prefix + EstimatePrecompileContractAddress,
+      nftAddress,
+    );
+    negativeScenarioVerification(tx, SERVER_ERROR);
+  });
+
+  //EGP-007
+  it('should call estimateGas with approve function without association - negative', async function () {
+    let amount = 1;
+    const tx = await contract.approveExternal.populateTransaction(tokenAddress, accounts[0].wallet.address, amount);
+    negativeScenarioVerification(tx, CALL_EXCEPTION);
+  });
+
+  //EGP-008
+  it('should call estimateGas with approveNFT function without association - negative', async function () {
+    const tx = await contract.approveNFTExternal.populateTransaction(nftAddress, accounts[1].wallet.address, 1);
+    negativeScenarioVerification(tx, CALL_EXCEPTION);
+  });
+
+  //EGP-009
+  it('should call estimateGas with dissociate token function for fungible token', async function () {
+    // Associate fungible token to account
+    estimateContractSigner0 = new ethers.Contract(
+      prefix + EstimatePrecompileContractAddress,
+      EstimatePrecompileContractJson.abi,
+      accounts[0].wallet,
+    );
+    await estimateContractSigner0.associateTokenExternal(
+      prefix + EstimatePrecompileContractAddress,
+      tokenAddress,
+      Constants.GAS.LIMIT_1_000_000,
+    );
+
+    const tx = await contract.dissociateTokenExternal.populateTransaction(estimateContractSigner0, tokenAddress);
+    const estimateGasResponse = await relay.call(RelayCalls.ETH_ENDPOINTS.ETH_ESTIMATE_GAS, [tx]);
+    baseGasCheck(estimateGasResponse, 785119);
```

Comment:
If the expected values change it will be hard to be maintained one by one, so you can use enum for them.